src/main/java/com/dnd/runus/application/challenge/ChallengeService.java

```diff
@@ -23,28 +25,43 @@ public class ChallengeService {
 
     private final RunningRecordRepository runningRecordRepository;
 
-    public List<ChallengesResponse> getChallenges(long memberId) {
+    public List<ChallengeWithCondition> getChallenges(long memberId) {
         OffsetDateTime todayMidnight = LocalDate.now(SERVER_TIMEZONE_ID)
                 .atStartOfDay(SERVER_TIMEZONE_ID)
                 .toOffsetDateTime();
         OffsetDateTime yesterday = todayMidnight.minusDays(1);
 
-        List<ChallengesResponse> challengesResponses;
+        List<RunningRecord> runningRecords =
+                runningRecordRepository.findByMemberIdAndStartAtBetween(memberId, yesterday, todayMidnight);
+
+        List<ChallengeWithCondition> allChallengesWithConditions =
+                new ArrayList<>(challengeRepository.findAllActiveChallengesWithConditions());
+
         // 어제 기록이 없으면
-        if (!runningRecordRepository.hasByMemberIdAndStartAtBetween(memberId, yesterday, todayMidnight)) {
-            challengesResponses = challengeRepository.findAllIsNotDefeatYesterday().stream()
-                    .map(ChallengesResponse::from)
-                    .collect(Collectors.toList());
-        } else {
-            challengesResponses = challengeRepository.findAllChallenges().stream()
-                    .map(ChallengesResponse::from)
+        if (runningRecords.isEmpty()) {
+            allChallengesWithConditions = allChallengesWithConditions.stream()
+                    .filter(challengeWithCondition ->
+                            !challengeWithCondition.challenge().isDefeatYesterdayChallenge())
                     .collect(Collectors.toList());
         }
 
         // 랜덤으로 2개 리턴
         Random randomWithSeed = new Random(todayMidnight.toEpochSecond());
-        Collections.shuffle(challengesResponses, randomWithSeed);
+        Collections.shuffle(allChallengesWithConditions, randomWithSeed);
+        List<ChallengeWithCondition> randomChallengeWithConditions = allChallengesWithConditions.subList(0, 2);
+
+        if (!runningRecords.isEmpty()) {
+            for (ChallengeWithCondition challenge : randomChallengeWithConditions) {
+                if (challenge.challenge().isDefeatYesterdayChallenge()) {
+                    // 어제의 기록과 관련된 챌린지면, 챌린지 비교할 값(성공 유무를 위한 목표 값) 재등록
+                    challenge
+                            .conditions()
+                            .forEach(condition -> condition.registerComparisonValue(
+                                    condition.goalMetricType().getActualValue(runningRecords.getFirst())));
+                }
+            }
```

Comment:
`!runningRecords.isEmpty()`일 때 for 문이나 스트림 중에 사용하면 좋을 것 같은데, 스트림도 괜찮은 것 같아서 예시로 사용해봤어요

보시고 성희님이 더 읽기 쉬운 방식으로 고르시면 될 것 같아요!

```suggestion
            allChallengesWithConditions.stream()
                    .filter(challenge -> challenge.challenge().isDefeatYesterdayChallenge())
                    .forEach(challenge -> challenge.conditions().forEach(condition ->
                            condition.registerComparisonValue(
                                    condition.goalMetricType().getActualValue(runningRecords.getFirst())
                            )
                    ));
```

Reply:
위에서 재원님이 `if (runningRecords.isEmpty())`관해서 말한 대로 if else로 처리한 다음에 스트림으로 처리하는게 더 좋아보이네요!